x/superfluid/keeper/stake_test.go

```diff
@@ -1028,6 +1028,158 @@ func (suite *KeeperTestSuite) TestRefreshIntermediaryDelegationAmounts() {
+			[]bool{false},
+			[]sdk.Dec{sdk.NewDec(6000000)},
+		},
+		{
+			"with single validator and additional superfluid delegations and single undelegation",
+			[]stakingtypes.BondStatus{stakingtypes.Bonded},
+			[]superfluidDelegation{{0, 0, 0, 1000000}, {0, 0, 0, 1000000}},
+			[]sdk.Coin{sdk.NewCoin("gamm/pool/1", sdk.NewInt(900000))},
+			[]uint64{1},
+			[]bool{false},
+			[]sdk.Dec{sdk.NewDec(11000000)},
+		},
+		{
+			"with multiple validators and multiple superfluid delegations and multiple undelegations",
+			[]stakingtypes.BondStatus{stakingtypes.Bonded, stakingtypes.Bonded},
+			[]superfluidDelegation{{0, 0, 0, 1000000}, {1, 1, 0, 1000000}},
+			[]sdk.Coin{sdk.NewCoin("gamm/pool/1", sdk.NewInt(600000)), sdk.NewCoin("gamm/pool/1", sdk.NewInt(400000))},
+			[]uint64{1, 2},
+			[]bool{false, false},
+			[]sdk.Dec{sdk.NewDec(4000000), sdk.NewDec(6000000)},
+		},
+		{
+			"add unbonding validator",
+			[]stakingtypes.BondStatus{stakingtypes.Bonded, stakingtypes.Unbonding},
+			[]superfluidDelegation{{0, 0, 0, 1000000}, {1, 1, 0, 1000000}},
+			[]sdk.Coin{sdk.NewCoin("gamm/pool/1", sdk.NewInt(600000)), sdk.NewCoin("gamm/pool/1", sdk.NewInt(400000))},
+			[]uint64{1, 2},
+			[]bool{false, false},
+			[]sdk.Dec{sdk.NewDec(4000000), sdk.NewDec(6000000)},
+		},
+		{
+			"add unbonded validator",
+			[]stakingtypes.BondStatus{stakingtypes.Bonded, stakingtypes.Unbonded},
+			[]superfluidDelegation{{0, 0, 0, 1000000}, {1, 1, 0, 1000000}},
+			[]sdk.Coin{sdk.NewCoin("gamm/pool/1", sdk.NewInt(600000)), sdk.NewCoin("gamm/pool/1", sdk.NewInt(400000))},
+			[]uint64{1, 2},
+			[]bool{false, false},
+			[]sdk.Dec{sdk.NewDec(4000000), sdk.NewDec(6000000)},
+		},
+		{
+			"undelegating not available lock id",
+			[]stakingtypes.BondStatus{stakingtypes.Bonded},
+			[]superfluidDelegation{{0, 0, 0, 1000000}},
+			[]sdk.Coin{sdk.NewCoin("gamm/pool/1", sdk.NewInt(600000))},
+			[]uint64{2},
+			[]bool{true},
+			[]sdk.Dec{},
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+
+			bondDenom := suite.App.StakingKeeper.GetParams(suite.Ctx).BondDenom
+
+			// setup validators
+			valAddrs := suite.SetupValidators(tc.validatorStats)
+
+			denoms, _ := suite.SetupGammPoolsAndSuperfluidAssets([]sdk.Dec{sdk.NewDec(20), sdk.NewDec(20)})
+
+			// setup superfluid delegations
+			_, intermediaryAccs, _ := suite.setupSuperfluidDelegations(valAddrs, tc.superDelegations, denoms)
+			suite.checkIntermediaryAccountDelegations(intermediaryAccs)
+
+			for index, lockId := range tc.superUnbondingLockIds {
+				// get intermediary account
+				accAddr := suite.App.SuperfluidKeeper.GetLockIdIntermediaryAccountConnection(suite.Ctx, lockId)
+				intermediaryAcc := suite.App.SuperfluidKeeper.GetIntermediaryAccount(suite.Ctx, accAddr)
+				valAddr := intermediaryAcc.ValAddr
+
+				lock, err := suite.App.LockupKeeper.GetLockByID(suite.Ctx, lockId)
+				if err != nil {
+					lock = &lockuptypes.PeriodLock{}
+				}
```

Comment:
This is specifically for the "undelegating not available lock id" test case. We don't want it to fail early and we want to attempt to partial superfluid undelegate with a non existing lock rather than catching it early. I can add a logic branch specific for this test case though.

Reply:
Added logic branch here https://github.com/osmosis-labs/osmosis/pull/5162/commits/ff187c6ac84650a315b8f6dd765c62298e4f020c